File: frontend/src/types/financialIndependence.ts
export interface FinancialIndependencePoint {
  year: number
  net_worth: number
  monthly_passive_income: number
  target_reached: boolean
}


export interface FinancialIndependence {
  currency: string

  monthly_income_target: number
  annual_income_target: number
  withdrawal_rate_percent: number
  required_capital: number

  current_net_worth: number
  current_monthly_passive_income: number

  remaining_gap: number
  progress_percent: number

  years_to_goal: number | null
  current_age: number
  projected_age_at_goal: number | null

  monthly_contribution: number
  annual_return_percent: number
  annual_property_growth: number

  yearly_projection: (
    FinancialIndependencePoint[]
  )
}

File: backend/schemas/financial_independence.py
from pydantic import BaseModel


class FinancialIndependencePoint(
    BaseModel
):
    year: int
    net_worth: float
    monthly_passive_income: float
    target_reached: bool


class FinancialIndependenceResponse(
    BaseModel
):
    currency: str

    monthly_income_target: float
    annual_income_target: float
    withdrawal_rate_percent: float
    required_capital: float

    current_net_worth: float
    current_monthly_passive_income: float

    remaining_gap: float
    progress_percent: float

    years_to_goal: int | None
    current_age: int
    projected_age_at_goal: int | None

    monthly_contribution: float
    annual_return_percent: float
    annual_property_growth: float

    yearly_projection: list[
        FinancialIndependencePoint
    ]